_posts/2018-03-27-roboporo.md

```diff
@@ -0,0 +1,32 @@
+
+## The Concept
+The story of this project is one best told in parts, its original concept coming from a dare between myself and another computing security student. It was to be a race to see who could create a simple bot program quicker, and with both of us not being the strongest of coders, I was somewhat certain that I could both take home a win in our little dare, and morph its outcome into a major project. At first my idea was simple *"I'll just make a quick little bot that draws from a hardcoded state and displays a picture linked with it in chat"*, I thought, and as I set out to do just this, I ran into a few problems.
+
+The first problem I ran into was the choice of coding language to use. I'm all one for innovation and breaking out of one's own bounds, however with Java as an option, I figured my final project would be much better refined if I began to write in something I had something of a base in. With the platform chosen, my next hurdle was finding an API to utilize Discord's app interface, and after a rather randomization of choices, I settled on [Discord4J](https://github.com/Discord4J/Discord4J). This was the first time in my experience that I implemented external API through Maven in my Java code, and after a bit of tooling around and research, I managed to accomplish my first goal - the Discord bot that would spit out a picture on command... That wasn't quite impressive enough however.
+
+## Improvement and Failures
+While the fist part of the project may have taken less than a week to complete, the following took until today to hash out. Moving forward, I grew tired of having hard-coded pictures and variables, this frustration warranting the need for a configuration, or storage file instead. My solution to this involved [SnakeYAML](https://bitbucket.org/asomov/snakeyaml), a YAML utility able to parse YAML file types into java Objects, the perfect solution to my issue. Once the configuration file was set up, I expanded upon the functionality of the program, adding a list command, the ability to add and remove entries from the configuration file, and even throwing in a novelty /roll command to add non-present convenience to the servers I was on.
+
+My final problem for this project came in the way of multi-server support however. When I attempted to place this bot onto multiple servers, I learned that they all shared the same configuration file - causing something of an issue when people wanted to keep images private in their communities instead of having them spread around. My solution to this was the dynamic generation of configuration files, accomplished through the use of an excessive amount of HashMaps, and adding a link between a given Guild's unique ID and the configuration file it was responsable for. At this innovation, the bot worked well above and beyond what I could have hoped to have accomplished from the start, and I was satisfied with its progress.
```

Comment:
`s/way of/form of/`

Reply:
`s/excessive amount/excessive number/` - I also think this is a little too editorialized, but it's your call.
`s/responsable/responsible/`

Reply:
`s/At this innovation/With this feature/`